src/OrchardCore.Modules/OrchardCore.Users/Views/UserFields.Edit.cshtml

```diff
@@ -72,15 +74,58 @@
 
                 element.type = 'password';
             }
+            @* The generated password should be at least 12 characters in length. But, it should meet the minimum requirements *@
+            const strongPasswordLength = @(Math.Max(12, Math.Max(IdentityOptions.Value.Password.RequiredLength, IdentityOptions.Value.Password.RequiredUniqueChars)));
+            const allowed = {
+                uppers: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+                lowers: 'abcdefghijklmnopqrstuvwxyz',
+                digits: '0123456789',
+                symbols: '!@@#$%^&*'
+            };
+
+            const getRandomChar = (str) => str.charAt(Math.floor(Math.random() * str.length));
+            const shuffle = (str) => str.sort(function() { return 0.5 - Math.random() });
+            const meetUniqueRule = (str) => {
+                let uniqueCharacters = str.filter((item, i, ar) => ar.indexOf(item) === i);
+
+                return uniqueCharacters.length >= @(IdentityOptions.Value.Password.RequiredUniqueChars);
+            };
+
             function generatePassword() {
-                const chars = '0123456789abcdefghijklmnopqrstuvwxyz!@@#$%^&*()ABCDEFGHIJKLMNOPQRSTUVWXYZ';
-                let password = '';
-                for (var i = 0; i <= strongPasswordLength; i++) {
-                    var randomNumber = Math.floor(Math.random() * chars.length);
-                    password += chars.substring(randomNumber, randomNumber + 1);
+                let password = [];
+
+                @* At least one uppercase *@
```

Comment:
Can we use JavaScript comments if it's required

Reply:
These comments are just for the developers and not needed to be printed in the HTML. Is there a reason why we need the comments to print?

Reply:
Ok